src/util.rs

```diff
@@ -46,6 +46,20 @@ pub fn parse_timespan(time_str: &str) -> usize {
     }
 }
 
+/// Sleep for a specified duration, minus an time spent doing other things.
+pub async fn sleep_minus_drift(
```

Comment:
This is probably a great candidate for:

[#inline(always)]

as it makes sense to inline this to reduce also the effects of the function call on the drift itself.

Reply:
Reading https://doc.rust-lang.org/reference/attributes/codegen.html and https://internals.rust-lang.org/t/when-should-i-use-inline/598/3 it seems the general consensus is that the compiler does a good job of this automatically if/when it makes sense? Granted, in all cases it's just a suggestion, which the compiler may still ignore.

I'm happy to add if you're confident it's the right thing to do; it's outside my experience.

Reply:
Does the async nature of this function defeat the purpose of inlining it? We will nearly always sleep and await.

Reply:
That’s a very good point - inlining on async fns does actually not work, yet per https://github.com/rust-lang/rust/issues/63647, so let’s leave this as is.

RTBM